app/src/main/java/com/asfoundation/wallet/billing/vkpay/VkPaymentIABFragment.kt

```diff
@@ -0,0 +1,266 @@
+import dagger.hilt.android.AndroidEntryPoint
+import java.math.BigDecimal
+import javax.inject.Inject
+
+
+@AndroidEntryPoint
+class VkPaymentIABFragment : BasePageViewFragment(),
+  SingleStateFragment<VkPaymentIABState, VkPaymentIABSideEffect> {
+
+  private val viewModel: VkPaymentIABViewModel by viewModels()
+  private val binding by lazy { VkPaymentIabLayoutBinding.bind(requireView()) }
+
+  @Inject
+  lateinit var formatter: CurrencyFormatUtils
+
+  @Inject
+  lateinit var vkDataPreferencesDataSource: VkDataPreferencesDataSource
+
+  private lateinit var iabView: IabView
+  private var navigatorIAB: Navigator? = null
+
+  private val authVkCallback = object : VkClientAuthCallback {
+    override fun onAuth(authResult: AuthResult) {
+      vkDataPreferencesDataSource.saveAuthVk(authResult.accessToken)
+      viewModel.hasVkUserAuthenticated = true
+      startVkCheckoutPay()
+    }
+  }
+
+  private var observeCheckoutResults: VkCheckoutResultDisposable =
+    VkPayCheckout.observeCheckoutResult { _ -> handleCheckoutResult() }
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    check(context is IabView) { "Vk payment fragment must be attached to IAB activity" }
+    iabView = context
+  }
+
+  override fun onResume() {
+    super.onResume()
+    if (SuperappKit.isInitialized()) {
+      if (vkDataPreferencesDataSource.getAuthVk().isNullOrEmpty() && viewModel.hasVkUserAuthenticated) {
+        startVkCheckoutPay()
+      }
+    }
+  }
+
+
+  override fun onCreateView(
+    inflater: LayoutInflater, @Nullable container: ViewGroup?,
+    @Nullable savedInstanceState: Bundle?
+  ): View {
+    //Build Vk Pay SuperApp Kit
+    VkPayManager.initSuperAppKit(
+      BuildConfig.VK_APP_NAME,
+      BuildConfig.VK_CLIENT_SECRET,
+      requireContext(),
+      R.mipmap.ic_launcher,
+      BuildConfig.VK_SDK_APP_ID,
+      activity
+    )
+    VkClientAuthLib.addAuthCallback(authVkCallback)
+    navigatorIAB = IabNavigator(parentFragmentManager, activity as UriNavigator?, iabView)
+    return VkPaymentIabLayoutBinding.inflate(inflater).root
+  }
+
+  override fun onViewCreated(view: View, @Nullable savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    viewModel.collectStateAndEvents(lifecycle, viewLifecycleOwner.lifecycleScope)
+    bindArguments()
+    setupTransactionCompleteAnimation()
+  }
+
+  private fun bindArguments() {
+    if (requireArguments().containsKey(CURRENCY_KEY) &&
+      requireArguments().containsKey(TRANSACTION_DATA_KEY) &&
+      requireArguments().containsKey(AMOUNT_KEY) &&
+      requireArguments().containsKey(ORIGIN_KEY)
+    ) {
+      viewModel.getPaymentLink(
+        requireArguments().getParcelable(TRANSACTION_DATA_KEY)!!,
+        (requireArguments().getSerializable(AMOUNT_KEY) as BigDecimal).toString(),
+        requireArguments().getString(CURRENCY_KEY)!!,
+        requireArguments().getString(ORIGIN_KEY)!!,
```

Comment:
problem in this case the value is mandatory. If it is null I call showError, but if it doesn't I can't put any value to avoid generating a wrong transaction